src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexParser.cs

```diff
@@ -0,0 +1,1916 @@
+            {
+                return false;
+            }
+
+            if (_currentToken.Kind == RegexKind.BarToken)
+            {
+                return false;
+            }
+
+            if (_currentToken.Kind == RegexKind.CloseParenToken)
+            {
+                return consumeCloseParen;
+            }
+
+            return true;
+        }
+
+        private RegexExpressionNode ParsePrimaryExpressionAndQuantifiers(RegexExpressionNode lastExpression)
+        {
+            var current = ParsePrimaryExpression(lastExpression);
+            if (current.Kind == RegexKind.SimpleOptionsGrouping)
+            {
+                // Simple options (i.e. "(?i-x)" can't have quantifiers attached to them).
+                return current;
+            }
+
+            switch (_currentToken.Kind)
+            {
+                case RegexKind.AsteriskToken: return ParseZeroOrMoreQuantifier(current);
+                case RegexKind.PlusToken: return ParseOneOrMoreQuantifier(current);
+                case RegexKind.QuestionToken: return ParseZeroOrOneQuantifier(current);
+                case RegexKind.OpenBraceToken: return TryParseNumericQuantifier(current, _currentToken);
+                default: return current;
+            }
+        }
+
+        private RegexExpressionNode TryParseLazyQuantifier(RegexQuantifierNode quantifier)
+        {
+            if (_currentToken.Kind != RegexKind.QuestionToken)
+            {
+                return quantifier;
+            }
+
+            // Whitespace allowed after the question and the next sequence element.
+            return new RegexLazyQuantifierNode(quantifier,
+                ConsumeCurrentToken(allowTrivia: true));
+        }
+
+        private RegexExpressionNode ParseZeroOrMoreQuantifier(RegexPrimaryExpressionNode current)
+        {
+            // Whitespace allowed between the quantifier and the possible following ? or next sequence item.
+            return TryParseLazyQuantifier(new RegexZeroOrMoreQuantifierNode(current, ConsumeCurrentToken(allowTrivia: true)));
+        }
+
+        private RegexExpressionNode ParseOneOrMoreQuantifier(RegexPrimaryExpressionNode current)
+        {
+            // Whitespace allowed between the quantifier and the possible following ? or next sequence item.
+            return TryParseLazyQuantifier(new RegexOneOrMoreQuantifierNode(current, ConsumeCurrentToken(allowTrivia: true)));
+        }
+
+        private RegexExpressionNode ParseZeroOrOneQuantifier(RegexPrimaryExpressionNode current)
+        {
+            // Whitespace allowed between the quantifier and the possible following ? or next sequence item.
+            return TryParseLazyQuantifier(new RegexZeroOrOneQuantifierNode(current, ConsumeCurrentToken(allowTrivia: true)));
+        }
+
+        private RegexExpressionNode TryParseNumericQuantifier(
+            RegexPrimaryExpressionNode expression, RegexToken openBraceToken)
```

Comment:
Seems like this method is called only once, with `_currentToken` value for `openBraceToken`. Consider removing the parameter and instead asserting that `_currentToken` is an `OpenBraceToken` kind.

Reply:
#won'tfix